### ISSUE TITLE
Getting multiple notifications for individual emails

### ISSUE THREAD

**Issue by bmurphy1976** [Thomas101/wmail#140]
I seem to be getting a lot of bad notifications on OSX when multiple emails come in.  When one email comes in at a time, I usually see only one notification popup.  However, I just had about 40 alerts come in at once and received over 300 notifications before the app became mostly unresponsive and I restarted it.

Also, I've seen what appears to be old notifications re-posted when restarting the app on occasion.  I might only have a couple unread items in my inbox, I'll open the app, and then 30 seconds later I'll get 140+ notifications.

**Comment by Thomas101** [comment]
Hi, which version of the app are you using?

**Comment by bmurphy1976** [comment]
`v1.2.3`

**Comment by Thomas101** [comment]
Is it only when you restart the app that you see this or do you ever get it whilst the app is running? I think this might be related to #129.

**Comment by bmurphy1976** [comment]
Definitely happens while runninng.  I hadn't restarted it since at least yesterday when I was inundated with alerts this morning.

**Comment by Thomas101** [comment]
hmmm that's not great. How many accounts do you have and are they inbox or gmail?

**Comment by bmurphy1976** [comment]
OK, interesting question, and I have an unusual answer.  I have 4 accounts, however, two of them I login as *both* Inbox and Gmail (so a total of 4 gmail and 2 inbox tabs).  I have notifications turned off for the inbox tabs.  I only use the inbox tabs to snooze the occasional message, otherwise I spend most if my time in the Gmail tabs.  Its' definitely the Gmail tabs that are giving me the notifications (they all have the little read Gmail icon).

Basically:

```
Tab1: Gmail Account A
Tab2: Gmail Account B
Tab3: Gmail Account C
Tab4: Inbox Account A
Tab5: Inbox Account B
Tab6: Gmail Account D
```

**Comment by joshrickert** [comment]
I have been having the same issue since 1.2.2. Just one account, set to use Inbox. It's spamming me with the same ~15 notifications for my emails over and over until they pile into the hundreds. The application is unresponsive while this happens.

Restarting doesn't seem to fix it. It starts when I launch the app and continues as long as it's running.

One time I think I managed to wait out the "queue" and the app became responsive again after a few hundred notifications over 10 minutes or so.

The app isn't really usable at the moment due to this issue.

**Comment by bmurphy1976** [comment]
I seem to have crossed some threshold tonight.  I only have 22 unread emails in my inbox, but now when I start the app my CPU goes through the roof and the app locks up.  I haven't been able to use it for a couple hours now, even after force killing it a couple times.

**Comment by bmurphy1976** [comment]
I opened up gmail in my browser and did some cleanup on my inbox.  It's back to being annoying, but usable again.

**Comment by Thomas101** [comment]
Thanks for these. I've been managed t reproduce this so I'll get it fixed and issue a patch later.

**Comment by Thomas101** [comment]
Hey, this is fixed. You can check it out by [downloading the latest release](https://thomas101.github.io/wmail/download). Thanks! 😊 😊 😊

**Comment by bmurphy1976** [comment]
Awesome, thank you!  So far so good!
